opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -195,6 +195,9 @@ message Metric {
 // "StartTimeUnixNano" is ignored for all data points.
 message Gauge {
   repeated NumberDataPoint data_points = 1;
+
+  // subtype provides additional type information for gauges
+  optional GaugeSubtype subtype = 2;
```

Comment:
Is `optional` necessary here? The zero-value for the enum GAUGE_SUBTYPE_NONE matches the current behavior, right? Do receivers need to be able whether `subtype` field was set on the wire?

Reply:
```suggestion
  optional GaugeSubtype sub_type = 2;
```

Reply:
> Do receivers need to be able whether subtype field was set on the wire?

No, receivers don't need to be able to tell if it was set or not.

> Is optional necessary here? The zero-value for the enum GAUGE_SUBTYPE_NONE matches the current behavior, right? 

Thats correct.  I meant to ask for advice: Is it better to have an optional field without a NONE zero-value, or a non-optional field with a NONE zero-value?

Reply:
Optional fields are slower (at least in Go where they are pointed-to and must be heap-allocated). I would go with non-optional enum that has a zero-value of none.

Reply:
done, and I make all instances of "subtype" or "Subtype" two words (sub_type, sub-type, or SubType)